vulnerabilities/importers/nginx.py

```diff
@@ -106,89 +109,103 @@ def to_advisories(self, data):
                     continue
 
                 if "Vulnerable" in child:
-                    vulnerable_packages = self.extract_vuln_pkgs(child)
+                    aff_pkgs = extract_vuln_pkgs(child)
                     continue
 
-            advisories.append(
-                Advisory(
-                    vulnerability_id=cve_id,
+            # TODO: Change this after https://github.com/nexB/univers/issues/8 is fixed
+            purl = PackageURL(type="generic", name="nginx")
+            affected_packages = []
+            for pkg in aff_pkgs:
+                for fixed_version in fixed_package_versions:
+                    affected_packages.append(
+                        AffectedPackage(
+                            package=purl,
+                            affected_version_specifier=pkg.affected_version_specifier,
+                            fixed_version=fixed_version,
+                        )
+                    )
+
+            advisory_data.append(
+                AdvisoryData(
                     summary=summary,
-                    affected_packages=nearest_patched_package(vulnerable_packages, fixed_packages),
+                    vulnerability_id=cve_id,
+                    affected_packages=affected_packages,
+                    references=references,
+                    date_published=datetime.now(),  # TODO: put real date here
                 )
             )
 
-        return advisories
+        return advisory_data
 
-    def extract_fixed_pkgs(self, vuln_info):
-        vuln_status, version_info = vuln_info.split(": ")
-        if "none" in version_info:
-            return {}
 
-        raw_ranges = version_info.split(",")
-        version_ranges = []
-        for rng in raw_ranges:
-            # Eg. "1.7.3+" gets converted to VersionSpecifier.from_scheme_version_spec_string("semver","^1.7.3")
-            # The advisory in this case uses `+` in the sense that any version
-            # with greater or equal `minor` version satisfies the range.
-            # "1.7.4" satisifes "1.7.3+", but "1.8.4" does not. "1.7.3+" has same
-            # semantics as that of "^1.7.3"
+def extract_fixed_pkg_versions(vuln_info):
+    vuln_status, version_info = vuln_info.split(": ")
+    if "none" in version_info:
+        return {}
 
-            version_ranges.append(
-                VersionSpecifier.from_scheme_version_spec_string("semver", "^" + rng[:-1])
-            )
+    raw_ranges = version_info.split(",")
+    versions = []
+    for rng in raw_ranges:
+        # Eg. "1.7.3+" gets converted to SemVersion(1.7.3)
+        # The way this needs to be interpreted is unique for nginx advisories
+        # More: https://github.com/nexB/vulnerablecode/issues/553
 
-        valid_versions = find_valid_versions(
-            self.version_api.get("nginx/nginx").valid_versions, version_ranges
-        )
+        versions.append(SemverVersion(rng.partition("+")[0].strip()))
```

Comment:
```suggestion
        # TODO: create a version scheme that's specific to nginx... because this is not exactly semver
        versions.append(SemverVersion(rng.partition("+")[0].strip()))
```